Runner/Utils.py

```diff
@@ -0,0 +1,278 @@
+    tasks.
+    """
+    blob_name = os.path.basename(file_path)
+
+    print('Uploading file {} to container [{}]...'.format(file_path,
+                                                          container_name))
+
+    block_blob_client.create_blob_from_path(container_name,
+                                            blob_name,
+                                            file_path)
+
+    sas_token = block_blob_client.generate_blob_shared_access_signature(
+        container_name,
+        blob_name,
+        permission=azureblob.BlobPermissions.READ,
+        expiry=datetime.datetime.utcnow() + datetime.timedelta(hours=2))
+
+    sas_url = block_blob_client.make_blob_url(container_name,
+                                              blob_name,
+                                              sas_token=sas_token)
+
+    return batchmodels.ResourceFile(file_path=blob_name,
+                                    blob_source=sas_url)
+
+async def wait_for_tasks_to_complete(batch_service_client, job_id, timeout):
+    """
+    Returns when all tasks in the specified job reach the Completed state.
+
+    :param batch_service_client: A Batch service client.
+    :type batch_service_client: `azure.batch.BatchServiceClient`
+    :param str job_id: The id of the job whose tasks should be to monitored.
+    :param timedelta timeout: The duration to wait for task completion. If all
+    tasks in the specified job do not reach Completed state within this time
+    period, an exception will be raised.
+    """
+    timeout_expiration = datetime.datetime.now() + timeout
+    print(timeout, timeout_expiration)
+
+    #print("Monitoring all tasks for 'Completed' state, timeout in {}...".format(timeout), end='')
+
+    while datetime.datetime.now() < timeout_expiration:
+        tasks = batch_service_client.task.list(job_id)
+
+        incomplete_tasks = [task for task in tasks if
+                            task.state != batchmodels.TaskState.completed]
+        if not incomplete_tasks:
+            return True, "job: {} successfully completed.".format(job_id)
+        else:
+            print("job: {} is running".format(job_id))
+            await asyncio.sleep(1)
+
+    return False, ValueError("ERROR: Tasks did not reach 'Completed' state within "
+                       "timeout period of " + str(timeout))
+
+async def check_task_output(batch_service_client, job_id, expected_output):
+    """Prints the stdout.txt file for each task in the job.
+
+    :param batch_client: The batch client to use.
+    :type batch_client: `batchserviceclient.BatchServiceClient`
+    :param str job_id: The id of the job with task output files to print.
+    """
+    
+    #print('Printing task output...')
+
+    tasks = batch_service_client.task.list(job_id)
+
+    for task in tasks:    
+        all_files = batch_service_client.file.list_from_task(job_id, task.id, recursive=True)
+        for f in all_files:
+            if expected_output in f.name:
```

Comment:
💡 Int his context is `f.name` a list or just `string with finlename` if latter is the case then you can just do `==` instead of `in` use.

Reply:
need to use in since we to deal with the filepath in the file